bin/makeLinkTrackletsInput_byNight.py

```diff
@@ -76,63 +74,62 @@ def writeDetsAndIds(detsFile, idsFile, compatibleNights):
     print "done with ", compatibleNights
     print ""
 
-
 if __name__=="__main__":
 
-    WINDOW_SIZE=int(sys.argv[1])
-    print "Reading MITI files from %s" %(INPUT_DETS_DIR)
-    print "Reading tracklet files from %s" %(INPUT_TRACKLETS_DIR)
-    print "Will write %d day window *.dets and *.ids files to %s" %(WINDOW_SIZE, OUTPUT_DIR)
+    import argparse
 
-    # Find the list of all *miti (OR INPUT_DETS_SUFFIX) files in the INPUT_DETS_DIR.
-    mitiFiles = glob.glob(os.path.join(INPUT_DETS_DIR + '*' + INPUT_DETS_SUFFIX))
+    parser = argparse.ArgumentParser(description="Combines nightly DiaSource files into detection and tracklet files covering a window of nights.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("diasDir", type=str, help="Directory containing DiaSources (detections) seperated in files by night.")
+    parser.add_argument("trackletDir", type=str,
+                        help="Directory containing the tracklet files to be used for linking.")
+    parser.add_argument("outputDir", type=str,
+                        help="Directory to put output detection and ids files in.")
+    parser.add_argument("--windowSize", type=int, default=WINDOW_SIZE,
+                        help="The number of nights in a window.")
+    parser.add_argument("--diasSuffix", type=str, default=INPUT_DETS_SUFFIX,
+                        help="Dias file suffix.")
+    parser.add_argument("--trackletSuffix", type=str, default=INPUT_TRACKLETS_SUFFIX,
+                        help="Tracklet file suffix.")
 
-    # Sort the files in MJD order.
-    nights = sorted(map(lambda x: int(x[len(INPUT_DETS_DIR):-(len(INPUT_DETS_SUFFIX))]), mitiFiles))
+    args = parser.parse_args()
 
+    print "Reading dias files from %s" % (args.diasDir)
+    print "Reading tracklet files from %s" % (args.trackletDir)
+    print "Will write %d day window *.dets and *.ids files to %s" % (args.windowSize , args.outputDir)
 
-    print "All nights: ", nights
+    # Find the list of all detection files
+    diasFiles = glob.glob(os.path.join(args.diasDir + '*' + args.diasSuffix))
 
+    # Sort the files in MJD order.
+    nights = sorted(map(lambda x: int(x[len(args.diasDir):-(len(args.diasSuffix))]), diasFiles))
```

Comment:
@mjuric @timj Considering your earlier comments, would it be appropriate for me to change this as well? To something like:

`nights = sorted(int(os.path.basename(x).split('.')[0]) for x in diasFiles)`